bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindAndReplaceMessageStatus.java

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Vector Informatik GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.ui.texteditor;
+
+/**
+ * Use by FindAndReplace to signal warnings, errors and messages to
+ * FindAndReplaceDialog and FindAndReplaceOverlay.
+ *
+ * @since 3.17
+ */
+class FindAndReplaceMessageStatus {
+	private boolean error;
+	private boolean warning;
+	private String message;
+
```

Comment:
I would be good to have some documentation on the meaning of the status values. Maybe this could be done via a constructor that explains its parameters or the field getters.